Fabric.Terminology.API/Scripts/Terminology-Intall-Utilities.psm1

```diff
@@ -35,74 +137,156 @@ function Get-TerminologyConfig {
     $appSettingsJson.BaseTerminologyEndpoint = $config.applicationEndpoint
-    $appSettingsJson.TerminologySqlSettings.ConnectionString = $config.sqlServerAddress
-    $appSettingsJson.SwaggerRootBasePath = $config.appName
+    $appSettingsJson.TerminologySqlSettings.ConnectionString = "Data Source=$($config.sqlServerAddress);Initial Catalog=$($config.sharedDbName); Trusted_Connection=True;"
     $appSettingsJson.IdentityServerSettings.ClientSecret = $config.appName
     $appSettingsJson.DiscoveryServiceClientSettings.DiscoveryServiceUrl = $config.discoveryServiceUrl
-    $appSettingsJson.ApplicationInsightsSettings.InstrumentationKey = "???"
+    $appSettingsJson.ApplicationInsightsSettings.InstrumentationKey = $config.appInsightsKey
     $appSettingsJson.ApplicationInsightsSettings.Enabled = $FALSE
 
-    $appSettingsJson | ConvertTo-Json | Set-Content $appSettings
+    $appSettingsJson | ConvertTo-Json -Depth 10 | Set-Content $appSettings
 }
 
 function Update-DiscoveryService() {
     param(
-        [TerminologyConfig] $config
+        [PSCustomObject] $config
     )
 
-    #$appDirectory = [System.IO.Path]::Combine($webroot, $appName)
+    Import-Module WebAdministration
+
+    $webroot = Get-WebFilePath -PSPath "IIS:\Sites\$($config.siteName)\$($config.appName)"
+    $terminologyAssembly = [System.IO.Path]::Combine($webroot, "Fabric.Terminology.API.dll")
+    $version = [System.Diagnostics.FileVersionInfo]::GetVersionInfo($terminologyAssembly).FileMajorPart
 
     $discoveryPostBody = @{
-        buildVersion   = "0.1"# TODO: [System.Diagnostics.FileVersionInfo]::GetVersionInfo("$appDirectory\Fabric.Terminology.API.dll").FileVersion;
-        serviceName    = "TerminologyService";
-        serviceVersion = 1;
-        friendlyName   = "Fabric.Terminology";
-        description    = "The Fabric.Terminology Service provides shared healthcare terminology data.";
-        serviceUrl     = $config.applicationEndpoint;
+        serviceName    = "TerminologyService"
+        serviceVersion = $version
+        friendlyName   = "Fabric.Terminology"
+        description    = "The Fabric.Terminology Service provides shared healthcare terminology data."
+        serviceUrl     = "http://localhost/TerminologyService/v$version"# TODO: replace with application endpoint
     }
     Add-DiscoveryRegistration $config.discoveryServiceUrl $config.iisUserCredentials $discoveryPostBody
 }
 
-function Publish-TerminologyDatabaseUpdates {
-    <# TODO: Ben create terminology db through dacpac
-    - add the following to Fabric.Identity.InstallPackage.targets: https://github.com/HealthCatalyst/Fabric.Identity/blob/master/Fabric.Identity.API/scripts/Fabric.Identity.InstallPackage.targets
-    #>
+function Publish-TerminologyDatabaseRole() {
+    param(
+        [PSCustomObject] $config,
+        [String] $DatabaseName,
+        [String] $RoleName
+    )
+
+    $Query = "DECLARE @cmd nvarchar(max)
```

Comment:
Somewhat trivial but I thought the casing convention was to use capitalized function parameters but lowercased local variables

Reply:
I will fix this, good point. I will also fix the bad spelling of the file name